zellij-server/src/panes/grid.rs

```diff
@@ -271,6 +271,23 @@ fn subtract_isize_from_usize(u: usize, i: isize) -> usize {
     }
 }
 
+macro_rules! dump_screen {
+    ($lines:expr) => {{
+        let mut is_first = true;
+        let mut buf = "".to_owned();
+
+        for line in &$lines {
+            if line.is_canonical && !is_first {
+                buf.push_str("\n");
+            }
+            let s: String = (&line.columns).into_iter().map(|x| x.character).collect();
+            buf.push_str(&(s.replace("[ ]*$", "")));
```

Comment:
I'm not sure I understand what's going on here...?

Reply:
I noticed that in some conditions, I can have a lot of white spaces until the end of the line. I'm removing those lines.

Reply:
So essentially delete whitespace at the end of the line? I think it might be there for when we need to color the background and such. Makes sense. Maybe add a comment to that effect?